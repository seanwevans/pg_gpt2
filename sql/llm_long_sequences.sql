-- Regression test for sequences longer than the legacy 1024-token block size.
SET client_min_messages = warning;

TRUNCATE llm_param,
         llm_dataset,
         llm_train_log,
         llm_tensor,
         llm_tensor_rt,
         llm_tensor_map,
         llm_tape;

-- Minimal parameters for a model with a single embedding dimension.
INSERT INTO llm_param(model, name, token_id, data, grad, m, v, step)
SELECT 'long_ctx', 'wte', token_id, pg_catalog.float4send(0::float4), NULL, NULL, NULL, 0
FROM generate_series(0, 1) AS token_id;

INSERT INTO llm_param(model, name, token_id, data, grad, m, v, step)
SELECT 'long_ctx', 'wpe', position, pg_catalog.float4send(position::float4), NULL, NULL, NULL, 0
FROM generate_series(0, 2047) AS position;

-- Insert a dataset row covering 2048 timesteps.
INSERT INTO llm_dataset(tokens, target)
SELECT
    ARRAY(SELECT 0 FROM generate_series(1, 2048)),
    ARRAY(SELECT 0 FROM generate_series(1, 2048));

SELECT array_length(tokens, 1) AS token_count
FROM llm_dataset
ORDER BY id
LIMIT 1;

SELECT octet_length(llm_embed(tokens, 'long_ctx', 1)) AS embed_bytes
FROM llm_dataset
ORDER BY id
LIMIT 1;

TRUNCATE llm_param,
         llm_dataset,
         llm_train_log,
         llm_tensor,
         llm_tensor_rt,
         llm_tensor_map,
         llm_tape;
