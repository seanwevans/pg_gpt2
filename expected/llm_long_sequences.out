SET
TRUNCATE TABLE
INSERT 0 2
INSERT 0 2048
INSERT 0 1
 token_count 
-------------
        2048
(1 row)

 embed_bytes 
-------------
        8192
(1 row)

TRUNCATE TABLE
